src/dialogs/azureCode.js

```diff
@@ -1,10 +1,44 @@
+var builder = require('botbuilder')
+var helper = require('../services/helper.js')
+
 module.exports = function (bot) {
   bot.dialog('azureCode', [
     function (session, args, next) {
-      session.send('Hello, I can help you get Azure Codes!')
-
-      // remember to ask here since async dialog
-      session.replaceDialog('isSatisfied')
+      builder.Prompts.text('Lets get you an Azure Code! First what is your full name?')
+    },
+    function (session, results) {
+      session.dialogData.studentName = results.response
+      builder.Prompts.text('Please tell me your email. This is needed so that your Azure code can be sent to you.')
+    },
+    function (session, results) {
+      session.dialogData.studentEmail = results.response
+      builder.Prompts.text('What is the name of your project?')
+    },
+    function (session, results) {
+      session.dialogData.projectName = results.response
+      builder.Prompts.text('Lastly, please give a brief description of your project.')
+    },
+    function (session, results) {
+      session.dialogData.projectDescription = results.response
+      return getAzureCode(session.dialogData)
+      .then(azureCode => {
+        if (azureCode) {
+          session.send('Your Azure Code has been sent to your email.')
```

Comment:
just fyi, we should send the email **before** the bot tells the user that the email is sent.